crates/swc_ecma_minifier/src/compress/optimize/inline.rs

```diff
@@ -651,31 +652,8 @@ where
                         None
                     }
                 })
-                .and_then(|v| {
-                    // Prevent infinite recursion.
-                    let ids = idents_used_by(&**v);
-                    if ids.contains(&i.to_id()) {
-                        None
-                    } else {
-                        Some(v)
-                    }
-                })
                 .cloned()
             {
-                match &*value {
-                    Expr::Lit(Lit::Num(..)) => {
```

Comment:
Is this really useless?

Reply:
If it's inlined in assign LHS, then it should be reassigned so it won't be inlined at first place.

Also if it is necessary, then why do string literal not need this check?

Reply:
Not sure, I think this code is wrong anyway